### PR TITLE
Add requirements for ClickHouse connector

### PR DESCRIPTION
**Pull request by jhlodin**
Add standard requirements section for the ClickHouse connector doc.

### CODE REVIEW

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 1]
```diff
@@ -7,6 +7,15 @@ The ClickHouse connector allows querying tables in an external
 query data in the databases on that instance, or combine it with other data
 from different catalogs accessing ClickHouse or any other supported data source.
 
+Requirements
+------------
+
+To connect to a ClickHouse instance, you need:
+
+* ClickHouse version 21.6 or higher.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/clickhouse.rst:15]: We test with 20.8. See https://github.com/trinodb/trino/blob/master/plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingClickHouseServer.java#L29

Comment (jhlodin) [reply docs/src/main/sphinx/connector/clickhouse.rst]: Got it, thanks. Updated.

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 2]
```diff
@@ -7,6 +7,15 @@ The ClickHouse connector allows querying tables in an external
 query data in the databases on that instance, or combine it with other data
 from different catalogs accessing ClickHouse or any other supported data source.
 
+Requirements
+------------
+
+To connect to a ClickHouse instance, you need:
+
+* ClickHouse version 20.8 or higher.
+* Network access from the Trino coordinator and workers to the ClickHouse
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/clickhouse.rst:16]: check in the clickhouse docs if the wording is right here 

options beyond the current "instance" imho are

* and workers to Clickhouse.
* and workers to the ClickHouse cluster.
* and workers to the Clickhouse server.

**Review by mosabua — COMMENTED** [review]
Trino PRs require on commit per PR .. please rebase/squash the commits in this PR. And ping me if you need help

**Review by m57lyra — APPROVED** [review]
Other than Manfred's point, LGTM

File: docs/src/main/sphinx/connector/clickhouse.rst (hunk 2 above)
Comment (jhlodin) [reply docs/src/main/sphinx/connector/clickhouse.rst:16]: Their docs refer to both ClickHouse servers and clusters. Updated accordingly.

**Comment by jhlodin** [comment]
> Trino PRs require on commit per PR .. please rebase/squash the commits in this PR. And ping me if you need help

Resolved

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 3]
```diff
@@ -28,8 +37,8 @@ appropriate for your setup:
 Multiple ClickHouse servers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-If you have multiple ClickHouse instances you need to configure one catalog for
-each instance. To add another catalog:
+If you have a cluster with multiple ClickHouse servers you need to configure one
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/clickhouse.rst:40]: This suggests that we support the "cluster" deployment of ClickHouse - which we don't know if it works. I think the earlier wording simply meant that if you have multiple ClickHouse servers each one needs their own catalog files.

Comment (jhlodin) [reply docs/src/main/sphinx/connector/clickhouse.rst]: Got it. Fixed.

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
FYI I created https://github.com/trinodb/trino/issues/8391 so that we can update the docs again about the "cluster" setup.

**Review by electrum — APPROVED** [review]

**Pull request merged**